### PR TITLE
add positional arguments

### PR DESCRIPTION
**Pull request by vmstarchenko**
add class Args

### CODE REVIEW

**Comment by shmuelamar** [comment]
Hi @vmstarchenko and thanks for your PR! I'll soon add AUTHORS.md so you can add yourself to :)
I still did not have enough time to review it deeply but i'll be glad if you can add unit tests for the new functionality as a start.

File: cbox/__main__.py [hunk 1]
```diff
@@ -6,7 +6,8 @@
 import cbox
 from cbox import concurrency
 
-__all__ = ('get_inline_func', 'main', )
+
+__all__ = ('get_inline_func', 'main')
```
Comment (reutsharabani) [inline cbox/__main__.py:10]: Please keep the extra comma, apparently it matters in python:


    >>> type((1))
    <class 'int'>

    >>> type((1,))
    <class 'tuple'>

File: cbox/cliparser.py [hunk 2]
```diff
@@ -13,29 +13,48 @@
...
+        self.names = names
+        self.type_ = type_
+        self.positional = positional
+        self.help_ = help_
+
+    def get_names(self):
+        if self.positional:
+            return self.names
+
+        return list(map(lambda _: ('--' if len(_) > 1 else '-') + _,
```
Comment (reutsharabani) [inline cbox/cliparser.py:28]: I think naming the lambda will make it easier to read and modify (these things tend to get more complex).

Maybe `prefix_flag`?

File: cbox/cliparser.py [hunk 3]
```diff
@@ -107,4 +124,13 @@ def _param2args(param, doc_param=None):
     else:
         arg_type = str
 
-    return arg_name, arg_type, arg_default, arg_required, arg_help
+    if not isinstance(arg_type, Arg):
+        arg_type = Arg(type_=arg_type)
+
+    if arg_type.names is None:
+        arg_type.names = [param.name.replace('_', '-')]
```
Comment (reutsharabani) [inline cbox/cliparser.py:131]: Maybe add both options? (`-` and `_`)?

    arg_type.names = [param.name, param.name.replace('_', '-')]

**Pull request closed without merging**